hivemind/client/moe.py

```diff
@@ -87,6 +87,9 @@ def forward(self, input: torch.Tensor, *args: torch.Tensor, **kwargs: torch.Tens
                 self._expert_info = next((expert.info for experts_i in chosen_experts for expert in experts_i))
             except grpc.RpcError as e:
                 logger.warning(f"Failed to get RemoteMixtureOfExperts.output_shape: {e}")
+            except StopIteration:
```

Comment:
I can also make this check more explicit and verify that chosen_experts isn't empty. Tell me if you prefer any specific way of handling this situation